### ISSUE TITLE
[Use case] XML multi-namespace resource 

### ISSUE THREAD

**Issue by ericprud** [ProfileNegotiation/I-D-Accept--Schema#21]
Consistent with XML's schema's addressing requirements for multi-namespace documents, the grammar allows for `profile-values` to have n pairs of XML namespace and schema location:
```
profile-value =  "<" URI-reference ">" [weight] |
                "<" URI-reference "0x20" URI-reference
                        *("0x20" URI-reference "0x20" URI-reference) ">"
```
While XML itself doesn't provide a semantics for combining documents, envelope languages (e.g. Atom, SOAP, and to some degree, RSS 2.0) are specifically designed to embed resources from other namespaces.

An HTTP request could prefer a syndication format of Atom with embedded content in HTML:
```
Accept-Profile: <
  http://www.w3.org/2005/Atom https://tools.ietf.org/html/rfc4287#appendix-B
  http://www.w3.org/1999/xhtml https://www.w3.org/2002/08/xhtml/xhtml1-strict.xsd>
```
and get bakd an Atom response containing those two namespaces:
```
<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">
  <title type="text">dive into mark</title>
  <id>tag:example.org,2003:3</id>
  <link rel="self" type="application/atom+xml" 
   href="http://example.org/feed.atom"/>
  <entry>
    <title>Atom draft-07 snapshot</title>
    <link rel="enclosure" type="audio/mpeg" length="1337"
     href="http://example.org/audio/ph34r_my_podcast.mp3"/>
    <id>tag:example.org,2003:3.2397</id>
    <content type="xhtml" xml:lang="en" 
     xml:base="http://diveintomark.org/">
      <div xmlns="http://www.w3.org/1999/xhtml">
        <p><i>[Update: The Atom draft is finished.]</i></p>
      </div>
    </content>
  </entry>
</feed>
```

**Comment by larsgsvensson** [comment]
This is a valid use case, and I think we need to be careful not to conflate profiles and schemas.

In my view a profile is technology-independent or at least media-type-independent, whereas a schema is an implementation of a profile geared towards a specific technology, e. g. the profile would say that you need an ID and a title in your data. Then for each media type there can be a schema document you can validate against, e. g. for `application/xml` data there is an XML Schema and for RDF data there is a ShEx document you can validate against. The XML Schema document and the ShEx document would be schemas in this parlance.

**Comment by RubenVerborgh** [comment]
I agree that we should not conflate profiles and schemas, but is there any reason to treat them differently for negotiation purposes?

**Comment by larsgsvensson** [comment]
> I agree that we should not conflate profiles and schemas, but is there any reason to treat them differently for negotiation purposes?

You mean that since they are only opaque URIs anyway it doesn't matter what the URIs represent? Yeah, could work...

**Comment by ericprud** [comment]
+1 to using profile for both.

I believe that for common cases, a document can range over one of the following:
1. `schema`: closed content model, e.g. a typical XML or JSON app, or a JSON-LD which is useful as JSON. In JSON-LD, this is also called a `frame`.
2. `vocabulary`: any generic (graph) representation like Turtle, OWL (any dialect including XML) or JSON-LD expanded form allows you to swap in different e.g. property and class names with the same semantics.

I propose that we use `profile` to identify either of these. While a single profile can identify a schema, we may want multiple profiles to identify the permutations of vocabularies in a resource. If a schema emerges which has no media type and is sufficiently flexible that one could plug different vocabularies in it, one could still use multiple profile directives to describe its scope of representations.

We slightly undermine the value of the [`+xml`](https://tools.ietf.org/html/rfc7303#section-4.2) and `+json` media types if folks can invent a standard profile URL instead. This is arguably a good thing as a media type registry that included all useful XML or JSON schemas would be ungainly.

**Comment by RubenVerborgh** [comment]
Seems that a `vocabulary` is basically a looser constraint than a `schema`. But yes, let's use `profile` for both, it's a very generic term.

> We slightly undermine the value of the +xml and +json media types 

On purpose. They have no official semantics. See https://ruben.verborgh.org/articles/fine-grained-content-negotiation/#possible-but-inadequate-workarounds
